### PR TITLE
[SPARK-38720][SQL][TESTS] Test the error class: CANNOT_CHANGE_DECIMAL_PRECISION

### PR DESCRIPTION
**Pull request by panbingkun**
## What changes were proposed in this pull request?
This PR aims to add a test for the error class CANNOT_CHANGE_DECIMAL_PRECISION to `QueryExecutionErrorsSuite`.

### Why are the changes needed?
The changes improve test coverage, and document expected error messages in tests.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By running new test:
```
$ build/sbt "sql/testOnly *QueryExecutionErrorsSuite*"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -400,4 +400,23 @@ class QueryExecutionErrorsSuite extends QueryTest
         "If necessary set spark.sql.ansi.enabled to false to bypass this error. ")
     }
   }
+
+  test("CANNOT_CHANGE_DECIMAL_PRECISION: cast string to decimal") {
+    withSQLConf(SQLConf.ANSI_ENABLED.key -> "true") {
+      val e = intercept[SparkArithmeticException] {
+        sql("select CAST('66666666666666.666' " +
+          "AS DECIMAL(8, 1))").collect()
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:408]: Should fit into one line?

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -400,4 +400,23 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+      assert(e.getErrorClass === "CANNOT_CHANGE_DECIMAL_PRECISION")
+      assert(e.getSqlState === "22005")
+      assert(e.getMessage ===
+        "Decimal(expanded,66666666666666.666,17,3}) cannot be represented as Decimal(8, 1). " +
+          "If necessary set spark.sql.ansi.enabled to false to bypass this error." +
+          """
+            |== SQL(line 1, position 7) ==
+            |select CAST('66666666666666.666' AS DECIMAL(8, 1))
+            |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            |""".stripMargin)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:419]: align this to the line above, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:408]: ok

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 3]
```diff
@@ -400,4 +400,22 @@ class QueryExecutionErrorsSuite extends QueryTest
...
+      assert(e.getErrorClass === "CANNOT_CHANGE_DECIMAL_PRECISION")
+      assert(e.getSqlState === "22005")
+      assert(e.getMessage ===
+        "Decimal(expanded,66666666666666.666,17,3}) cannot be represented as Decimal(8, 1). " +
+        "If necessary set spark.sql.ansi.enabled to false to bypass this error." +
+          """
+            |== SQL(line 1, position 7) ==
+            |select CAST('66666666666666.666' AS DECIMAL(8, 1))
+            |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            |""".stripMargin)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:418]:
```suggestion
        """
          |== SQL(line 1, position 7) ==
          |select CAST('66666666666666.666' AS DECIMAL(8, 1))
          |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          |""".stripMargin)
```

**Comment by MaxGekk** [comment]
Waiting for https://github.com/panbingkun/spark/actions/runs/2184786158

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**